src/electionguard/decryption_mediator.py

```diff
@@ -42,81 +48,95 @@ class DecryptionMediator:
         default_factory=lambda: {}
     )
-    _missing_guardians: DataStore[MISSING_GUARDIAN_ID, ElectionPublicKey] = field(
-        default_factory=lambda: DataStore()
-    )
-
-    _decryption_shares: Dict[AVAILABLE_GUARDIAN_ID, TallyDecryptionShare] = field(
+    _missing_guardians: Dict[MISSING_GUARDIAN_ID, ElectionPublicKey] = field(
         default_factory=lambda: {}
     )
-    """
-    A collection of Decryption Shares for each Available Guardian
-    """
-
     _lagrange_coefficients: Dict[
         MISSING_GUARDIAN_ID, Dict[AVAILABLE_GUARDIAN_ID, ElementModQ]
     ] = field(default_factory=lambda: {})
     """
     A collection of lagrange coefficients `w_{i,j}` computed by available guardians for each missing guardian
     """
 
-    _compensated_decryption_shares: Dict[
-        MISSING_GUARDIAN_ID,
-        Dict[AVAILABLE_GUARDIAN_ID, CompensatedTallyDecryptionShare],
-    ] = field(default_factory=lambda: {})
-    """
-    A collection of Compensated Decryption Shares for each Available Guardian
-    """
-
-    def announce(self, guardian: Guardian) -> Optional[TallyDecryptionShare]:
+    def announce(
+        self, guardian: Guardian
+    ) -> Optional[Tuple[DecryptionShare, Dict[BALLOT_ID, DecryptionShare]]]:
         """
         Announce that a Guardian is present and participating in the decryption.
         A Decryption Share will be generated for the Guardian
 
         :param guardian: The guardian who will participate in the decryption.
-        :return: a `TallyDecryptionShare` for this `Guardian` or `None` if there is an error.
+        :return: decryption shares for tally and ballot for this `Guardian` or `None` if there is an error.
         """
 
         # Only allow a guardian to announce once
         if guardian.object_id in self._available_guardians:
             log_info(f"guardian {guardian.object_id} already announced")
-            return self._decryption_shares[guardian.object_id]
+            return (
+                self._tally_shares[guardian.object_id],
+                self._ballot_shares[guardian.object_id],
+            )
 
-        # Compute the Decryption Share for the guardian
-        share = compute_decryption_share(
+        # Compute the tally and ballot decryption shares
+        tally_share = compute_decryption_share(
             guardian, self._ciphertext_tally, self._encryption
         )
-        if share is None:
+        if tally_share is None:
             log_warning(
-                f"announce could not generate decryption share for {guardian.object_id}"
+                f"announce could not generate tally decryption share for {guardian.object_id}"
             )
             return None
+        self._tally_shares[guardian.object_id] = tally_share
 
-        # Submit the share
-        if self._submit_decryption_share(share):
-            self._available_guardians[guardian.object_id] = guardian
-        else:
+        ballot_shares = compute_decryption_share_for_ballots(
```

Comment:
maybe adda comment above this code block so it matches the pattern in the rest of the file?